### ISSUE TITLE
'Cannot start new display' error upon unlocking screensaver

### ISSUE THREAD

**Issue by ACinnamonUserInUK** [linuxmint/cinnamon-screensaver#175]
What I did:

- lock the screen using keyboard shortcut;
- clicked the 'change user' icon (is it the change user icon? My system has only one user that should and does appear on the login screen - and clicking this mysterious icon seemed to do nothing);
- entered the correct password and pressed return.

At that point a titleless window appeared, with this content: 'Cannot start new display // The X server failed. Perhaps it is not configured well.' See image below:

![bug](https://cloud.githubusercontent.com/assets/9998846/21297984/63f936b0-c581-11e6-8ecf-e5689e5b9002.png)

The window had a single button, marked 'OK'. After clicking it several times (once seem insufficient to do anything - or perhaps I just had to wait), all seemed back to normal.

It seems that it is the clicking of the mysterious ('change user'?) button that causes the trouble. Merely locking and unlocking does not cause the problem.

**Comment by Mintulix** [comment]
I had this too the first time after the upgrade to LM 18.1.
I rebootet once again (2nd time) and since then the error is gone. Perhaps just luck, I don't know yet, but please try a reboot to check if it's gone.

**Comment by ACinnamonUserInUK** [comment]
@Mintulix: no, I've rebooted again (having already rebooted when prompted to by the update process) and the problem persists. Also, that button does need labelling - but the urgent thing is to stop is causing this X server problem.

**Comment by mtwebster** [comment]
What do you use for a login manager? mdm?

**Comment by ACinnamonUserInUK** [comment]
Yes, I think so. I haven't changed it and the password prompter for Settings -> Login mentions 'mdmsetup'.

Some statistics about my system:

```
Kernel: 4.4.0-53-generic 
Shell: /bin/bash 
Resolution: 1280x800 
DE: Cinnamon 
WM: Mutter (Muffin) 
WM Theme: Cinnamon (SimpleX-Grayscale_NJ) 
Theme: Mint-Y [GTK2/3] 
Icons: Paper [GTK2/3] 
Font: Noto Sans UI 11 [GTK2/3] 
CPU: Intel 2 Duo T7500 (2) @ 2.2GHz 
GPU: Intel Integrated Graphics 
Memory: 1590MB / 7974MB 

Intel Corporation Mobile GM965/GL960 Integrated Graphics Controller
OpenGL Vendor:  Intel Open Source Technology Center
OpenGL Renderer:  Mesa DRI Intel(R) 965GM 
OpenGL Version:  2.1 Mesa 13.0.2 - padoka PPA
Client glx vendor:  Mesa Project and SGI
Client glx version:  1.4
Server glx vendor:  SGI
Server glx version:  1.4
GLX version:  1.4
OpenGLES PVS:  OpenGL ES 2.0 Mesa 13.0.2 - padoka PPA
OpenGLES PSLVS:  OpenGL ES GLSL ES 1.0.16
Kernel driver in use:  i915
Total Memory prefetchable: 1M - 
Total Memory non-prefetchable: 256M - 
```

**Comment by Mintulix** [comment]
I had this message again. Tried to switch the user (I have a second one) in the new unlock-screen a few times, but nothing happened. After login (my own user) an error-message window (see above) was shown as many times as I tried before. I don't know what triggers the failure (a previous suspend?), but sometimes it works as expected.

**Comment by mtwebster** [comment]
I'll try to re-create your crypt setup to reproduce when i get a chance.

**Comment by Mintulix** [comment]
Thanks, but I don't want to steal the thread. The original issue poster (ACinnamonUserInUK) didn't mention ecryptfs errors, this is just a thought of mine.

**Comment by ACinnamonUserInUK** [comment]
The newest version of the screensaver (the one that comes with, or accompanies, Cinnamon 3.2.7 - I believe) solves the problem - there is no more 'X server failed' message. That's good!

It remains the case that there is no indication, save the little picture itself, that the change user button is the change user button. Now, since clicking that button _logs the current user out_, I really do think that _the user should be given more of an indication of what it does / what is about to happen_. Even just giving a tooltip ('change user') would be a step in that direction, and I imagine it would be easy to implement and it would improve the user experience.

**Comment by mtwebster** [comment]
@Mintulix can you check the latest version of the screensaver as well (pretty sure @clefebvre did a point release today) and see if it still manifests for you?  I'm not that hopeful in your case but I'd like to be sure before messing with pam modules some more (I'm still having nightmares)

In the meantime I'll see about adding tooltips to some of the ui elements - it should be simple enough to do, and since it doesn't add to any clutter, I don't see why not.

**Comment by Mintulix** [comment]
V. 3.2.12 now. I still have the message "cinnamon-screensaver-pam-helper: pam_ecryptfs: seteuid error" in the log, but everything is ok with switching users and decryption of my ecryptfs-folder.
BUT I still don't know how to reproduce the "Cannot start new display"-message error, I had it three times in the last three days (and I played a lot with screensaver and the new settings), so may be it's not fixed yet. B.t.w - the call of the user switching has a significant delay of nearly five seconds until you get the mdm-login screen (with ssd) - no tragedy, but I think it was much faster in 17.3 (related to systemd?).

The next few days I will not have the time testing (and so should you all), but I'll report back if I have any issues like that. So no more nightmares please, thank you very much for your work through all the years and happy holiday to you.

**Comment by Mintulix** [comment]
Sorry to say - for me, it's still not fixed, too. I have three users now and switched sucessfully a few times between two of them, both with screensaver enabled (and using them because of long interruptions of my work). Sometimes I switch using the user-applet, sometimes when the screen is locked I switch using the new round "switch-user"-button in lock screen.
First issue: after some switching one user got locked, i.e. there was no unlock-screen, just the clock on dimmed background (discribed in another issue).
Second issue: I tried to start the console with ctrl+alt+F1...F4 to kill xscreensaver (that worked once), but just got a black screen. Changing back to X worked, I had the unlock-screen for my main user then.
Third issue: Trying to switch users from that unlock-screen led to nothing but a lot of "Cannot start new display"-messages after login. There is no (graphical) way to switch users again until reboot.

**Comment by mtwebster** [comment]
What happens if you just open a terminal and run:

`mdmflexiserver --startnew Standard`

**Comment by Mintulix** [comment]
Output:
```
$ mdmflexiserver --startnew Standard
Found no MDM server, ordering a new one
Locking cinnamon-screensaver

```
MDM login screen is shown with both sessions still logged in, I can login to my main user (seems ok).
But my session had no errors until now, should I check the same when I have the issues described above?

**Comment by clefebvre** [comment]
Next time the "Cannot start new display" dialog appears.

Switch to console to kill the screensaver:

`killall cinnamon-screensaver`

Type:

`xprop`

and click on the window to get information about it.

Then type this to get a hierarchical list of processes:

`ps -ef --forest`

Then type this to get information about MDM displays:

`mdmflexiserver -c "ATTACHED_SERVERS" -a`

And the mdm version with:

`mdm --version`

Then finally, dismiss the dialog, and try to switch users manually with:

`mdmflexiserver`

**Comment by mtwebster** [comment]
Can you look at /var/auth.log also.

Also have a look at the following folders, do an `ls -la` on them:

`/lib/security` and `/lib/x86_64-linux-gnu/security` (or `/lib/i386-linux-gnu` for 32 bit)

Do any of your `pam_` files have permissions other than `-rw-r--r--` or owned by other than root?

**Comment by Mintulix** [comment]
I'm still watching this, but it didn't happen since my last entry. But I think there were no updates concerning this issue.
@ACinnamonUserInUK can you still reproduce it and collect the informations?

**Comment by ACinnamonUserInUK** [comment]
I do indeed still have the problem, I am afraid. For the following happens. Once the screensave has started, if I click on the 'change user' icon, and then enter my password (still for the original user - the only user the appears on the lock screen), I see the following.

![image](https://cloud.githubusercontent.com/assets/9998846/22230006/2fefa462-e1d3-11e6-96a1-a1a1fb189c8c.png)

I will edit this post once I've collected the log information.

**Comment by ACinnamonUserInUK** [comment]
More logs

I do not seem to have a /var/auth.log file.

/lib/security and /lib/x86_64-linux-gnu/security: yes, some items do have permissions other than the ones you specified. Those aberrant items and their permissions are as follows.

```
ls -la /lib/security
total 44
drwxr-xr-x  2 root root  4096 Jul 15  2016 .
drwxr-xr-x 26 root root  4096 Jan 21 01:32 ..
```

**Comment by Mintulix** [comment]
I have a clue! Got the error again with the following constellation:
The file /etc/X11/xorg.conf.d/20-intel.conf containing
```
Section "Device"
    Identifier "Intel Graphics"
    Driver "intel"
    Option "AccelMethod" "uxa"
EndSection
```
I also have dual graphics in my notebook, I don't know if that is critical for ths error:
```
$ inxi -G
Graphics:  Card-1: Intel 3rd Gen Core processor Graphics Controller
           Card-2: Advanced Micro Devices [AMD/ATI] Venus XT [Radeon HD 8870M / R9 M270X/M370X]
           Display Server: X.Org 1.18.4 driver: intel
           Resolution: 1920x1080@60.21hz
           GLX Renderer: Mesa DRI Intel Ivybridge Mobile
           GLX Version: 3.0 Mesa 11.2.0
```

Then I switched users a few times, after locking one user, standby, etc., could not switch users, but could login to my own user normally and saw the error message twice again.

And I hadn't have the error a long time without this file (I deleted it for testing purposes).
I don't know if 20-intel.conf was created during installation or later, but deleting this file _seems_ to fix the error.

Btw. I can't switch to AMD with "DRI_PRIME ...", unless I insert
`    Option "DRI" "3"`
into 20-intel.conf. Then the switch to AMD works, but I have worse errors switching users like black screens without console or any "blind console input effect" (just restart with Magic SysRQ works).

I also went through the instructions from Clem and Michael, but the logs were nearly the same as from  ACinnamonUserInUK.

**Comment by ACinnamonUserInUK** [comment]
> I don't know if 20-intel.conf was created during installation or later, but deleting this file seems to fix the error.

I have had the problem _without_ a `20-intel.conf` file on my system, _but_, when that was the case, I _did_ have some xorg.conf file. (I can't remember exactly what that latter file was called and subsequently I've deleted it.)

**Comment by duraz0rz** [comment]
Hi, I'm getting the same error on my desktop. I'm fairly new to running Linux full-time, but I'm not afraid of anything, so what do I need to run to get any pertinent info into your hands?

Relevant system specs:
Intel Core i7 6700K (with microcode drivers)
16GB RAM
Nvidia GTX 1070 (proprietary drivers)

**Comment by ACinnamonUserInUK** [comment]
Since creating a new user on my system (which had its own pecularities, which I may report separately), I no longer have the 'cannot start new display' error. However, the following does, now, happen: the 'change user' and whatever-the-other-one-is tooltips - the tooltips that got added - flicker on and off like crazy.

**Comment by Mintulix** [comment]
Tooltip flickering: see my comments from January here https://github.com/linuxmint/cinnamon-screensaver/issues/178

**Comment by duraz0rz** [comment]
I noticed from the comment @Mintulix made a week ago that he was having a problem with iGPU + dedicated GPU setup on a laptop. I have a similar setup, albeit on my desktop (I had my iGPU enabled alongside my GTX 1070). I decided to see what disabling the iGPU does, and the switch user problem goes away both on the lock screen and the normal switch user functionality.

**Comment by ACinnamonUserInUK** [comment]
@duraz0rz: Yet I and some other contributors to this thread have integrated graphics only . .
